azure-pipelines.yml

```diff
@@ -0,0 +1,189 @@
+# Starter pipeline
+# Start with a minimal pipeline that you can customize to build and deploy your code.
+# Add steps that build, run tests, deploy, and more:
+# https://aka.ms/yaml
+
+variables:
+  # The $(FlutterToolPath) variable is generated by the FlutterInstall task 
+  # See code in the Flutter Build task https://github.com/aloisdeniel/vsts-flutter-tasks/blob/master/tasks/build/index.ts
+  flutterExecPath: $(FlutterToolPath)/flutter.bat
+  versionNumberRegex: '(?<=version: ).*'
+
+trigger:
+  batch: 'true'
+  branches:
+    include:
+      - develop
+      - release
+      - hotfix
+      - master
+  paths:
+    include:
+      - device_calendar/*
+
+stages:
+  # ----------- CI -----------
+  - stage: Build
+    jobs: 
+     - job: BuildAndroidAndIos
+       pool:
+        vmImage: 'macOS-latest'
+       steps:
+        - task: FlutterInstall@0
+          displayName: 'Flutter install'
+          inputs:
+            channel: 'stable'
+            version: 'latest'
+
+        - task: FlutterBuild@0
+          displayName: 'Flutter build - Android'
+          inputs:
+            target: 'aab'
+            projectDirectory: 'device_calendar/example'
+
+        - task: FlutterBuild@0
+          displayName: 'Flutter build - iOS'
+          inputs:
+            target: 'ios'
+            projectDirectory: 'device_calendar/example'
+            iosCodesign: false
+            
+  # -----------CD Pre-release -----------
+  - stage: DevelopmentRelease
+    dependsOn: Build
+    condition: and(succeeded(), in(variables['Build.SourceBranch'], 'refs/heads/release', 'refs/heads/hotfix'))
```

Comment:
@mkieres @nickrandolph just came up with a thought that `Build.SourceBranch` isn't right. I believe it should be `Build.TargetBranch`.

When I want to pre-release a version, we're creating a PR from `develop` branch and merging into `release`. In this case, the source branch is `develop` and target branch is `release`.
If we use `Build.SourceBranch`, then the CD Pre-release will never run because the condition is looking for a source branch of `release` or `hotfix` where it's passing `develop`.

What you're thoughts?

Reply:
But little worried here if we use `Build.TargetBranch`, it might trigger the CD when I create a new PR or update something in a PR?

Reply:
I think SourceBranch is ok for the moment because it just means that this will trigger once the PR has been merged. We might eventually look at making this a pre-merge action but for now I think this is ok